Exception hiding the actual cause leads to unusable information

When files can not be parsed, I end up with the following exception, which hides the original one and does not give any information about the original failure or file being parsed.  Very hard to find out what the problem is.
```
[ERROR] Failed to execute goal org.openrewrite.maven:rewrite-maven-plugin:4.37.0:run (default-cli) on project components: Execution default-cli of goal org.openrewrite.maven:rewrite-maven-plugin:4.37.0:run failed: java.lang.IllegalStateException: Maven visitors should not be visiting XML documents without a Maven marker -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.openrewrite.maven:rewrite-maven-plugin:4.37.0:run (default-cli) on project components: Execution default-cli of goal org.openrewrite.maven:rewrite-maven-plugin:4.37.0:run failed: java.lang.IllegalStateException: Maven visitors should not be visiting XML documents without a Maven marker
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:375)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:294)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:960)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default-cli of goal org.openrewrite.maven:rewrite-maven-plugin:4.37.0:run failed: java.lang.IllegalStateException: Maven visitors should not be visiting XML documents without a Maven marker
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:148)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:370)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:294)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:960)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.openrewrite.internal.RecipeRunException: java.lang.IllegalStateException: Maven visitors should not be visiting XML documents without a Maven marker
    at org.openrewrite.TreeVisitor.visit (TreeVisitor.java:326)
    at org.openrewrite.TreeVisitor.visitAndCast (TreeVisitor.java:358)
    at org.openrewrite.xml.XmlVisitor.lambda$visitTag$2 (XmlVisitor.java:91)
    at org.openrewrite.internal.ListUtils.lambda$map$0 (ListUtils.java:141)
    at org.openrewrite.internal.ListUtils.map (ListUtils.java:123)
    at org.openrewrite.internal.ListUtils.map (ListUtils.java:141)
    at org.openrewrite.xml.XmlVisitor.visitTag (XmlVisitor.java:91)
    at org.openrewrite.maven.MavenIsoVisitor.visitTag (MavenIsoVisitor.java:38)
    at org.openrewrite.maven.MavenDownloadingExceptions$1.visitTag (MavenDownloadingExceptions.java:92)
    at org.openrewrite.maven.MavenDownloadingExceptions$1.visitTag (MavenDownloadingExceptions.java:66)
    at org.openrewrite.xml.tree.Xml$Tag.acceptXml (Xml.java:426)
    at org.openrewrite.xml.tree.Xml.accept (Xml.java:50)
    at org.openrewrite.TreeVisitor.visit (TreeVisitor.java:278)
    at org.openrewrite.TreeVisitor.visitAndCast (TreeVisitor.java:358)
    at org.openrewrite.xml.XmlVisitor.visitDocument (XmlVisitor.java:71)
    at org.openrewrite.maven.MavenIsoVisitor.visitDocument (MavenIsoVisitor.java:23)
    at org.openrewrite.maven.MavenIsoVisitor.visitDocument (MavenIsoVisitor.java:20)
    at org.openrewrite.xml.tree.Xml$Document.acceptXml (Xml.java:151)
    at org.openrewrite.xml.tree.Xml.accept (Xml.java:50)
    at org.openrewrite.TreeVisitor.visit (TreeVisitor.java:278)
    at org.openrewrite.TreeVisitor.visitNonNull (TreeVisitor.java:193)
    at org.openrewrite.maven.MavenDownloadingExceptions.warn (MavenDownloadingExceptions.java:94)
    at org.openrewrite.maven.UncheckedMavenDownloadingException.<init> (UncheckedMavenDownloadingException.java:27)
    at org.openrewrite.maven.MavenParser.parseInputs (MavenParser.java:132)
    at org.openrewrite.Parser.parse (Parser.java:41)
    at org.openrewrite.maven.MavenMojoProjectParser.parseMaven (MavenMojoProjectParser.java:309)
    at org.openrewrite.maven.MavenMojoProjectParser.listSourceFiles (MavenMojoProjectParser.java:119)
    at org.openrewrite.maven.AbstractRewriteMojo.listResults (AbstractRewriteMojo.java:209)
    at org.openrewrite.maven.AbstractRewriteRunMojo.execute (AbstractRewriteRunMojo.java:48)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:370)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:294)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:960)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: java.lang.IllegalStateException: Maven visitors should not be visiting XML documents without a Maven marker
    at org.openrewrite.maven.MavenVisitor.lambda$getResolutionResult$0 (MavenVisitor.java:67)
    at java.util.Optional.orElseThrow (Optional.java:403)
    at org.openrewrite.maven.MavenVisitor.getResolutionResult (MavenVisitor.java:67)
    at org.openrewrite.maven.MavenDownloadingExceptions$1.visitTag (MavenDownloadingExceptions.java:80)
    at org.openrewrite.maven.MavenDownloadingExceptions$1.visitTag (MavenDownloadingExceptions.java:66)
    at org.openrewrite.xml.tree.Xml$Tag.acceptXml (Xml.java:426)
    at org.openrewrite.xml.tree.Xml.accept (Xml.java:50)
    at org.openrewrite.TreeVisitor.visit (TreeVisitor.java:278)
    at org.openrewrite.TreeVisitor.visitAndCast (TreeVisitor.java:358)
    at org.openrewrite.xml.XmlVisitor.lambda$visitTag$2 (XmlVisitor.java:91)
    at org.openrewrite.internal.ListUtils.lambda$map$0 (ListUtils.java:141)
    at org.openrewrite.internal.ListUtils.map (ListUtils.java:123)
    at org.openrewrite.internal.ListUtils.map (ListUtils.java:141)
    at org.openrewrite.xml.XmlVisitor.visitTag (XmlVisitor.java:91)
    at org.openrewrite.maven.MavenIsoVisitor.visitTag (MavenIsoVisitor.java:38)
    at org.openrewrite.maven.MavenDownloadingExceptions$1.visitTag (MavenDownloadingExceptions.java:92)
    at org.openrewrite.maven.MavenDownloadingExceptions$1.visitTag (MavenDownloadingExceptions.java:66)
    at org.openrewrite.xml.tree.Xml$Tag.acceptXml (Xml.java:426)
    at org.openrewrite.xml.tree.Xml.accept (Xml.java:50)
    at org.openrewrite.TreeVisitor.visit (TreeVisitor.java:278)
    at org.openrewrite.TreeVisitor.visitAndCast (TreeVisitor.java:358)
    at org.openrewrite.xml.XmlVisitor.visitDocument (XmlVisitor.java:71)
    at org.openrewrite.maven.MavenIsoVisitor.visitDocument (MavenIsoVisitor.java:23)
    at org.openrewrite.maven.MavenIsoVisitor.visitDocument (MavenIsoVisitor.java:20)
    at org.openrewrite.xml.tree.Xml$Document.acceptXml (Xml.java:151)
    at org.openrewrite.xml.tree.Xml.accept (Xml.java:50)
    at org.openrewrite.TreeVisitor.visit (TreeVisitor.java:278)
    at org.openrewrite.TreeVisitor.visitNonNull (TreeVisitor.java:193)
    at org.openrewrite.maven.MavenDownloadingExceptions.warn (MavenDownloadingExceptions.java:94)
    at org.openrewrite.maven.UncheckedMavenDownloadingException.<init> (UncheckedMavenDownloadingException.java:27)
    at org.openrewrite.maven.MavenParser.parseInputs (MavenParser.java:132)
    at org.openrewrite.Parser.parse (Parser.java:41)
    at org.openrewrite.maven.MavenMojoProjectParser.parseMaven (MavenMojoProjectParser.java:309)
    at org.openrewrite.maven.MavenMojoProjectParser.listSourceFiles (MavenMojoProjectParser.java:119)
    at org.openrewrite.maven.AbstractRewriteMojo.listResults (AbstractRewriteMojo.java:209)
    at org.openrewrite.maven.AbstractRewriteRunMojo.execute (AbstractRewriteRunMojo.java:48)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:370)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:294)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:960)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
[ERROR] 
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
➜  camel git:(main) ✗ 
```
This looks related to #2334

Fwiw, the PR then leads to the following output:
```
Caused by: org.openrewrite.maven.UncheckedMavenDownloadingException: Failed to download dependencies for components/camel-iota/pom.xml:
<?xml version="1.0" encoding="UTF-8"?>
<!--

    Licensed to the Apache Software Foundation (ASF) under one or more
    contributor license agreements.  See the NOTICE file distributed with
    this work for additional information regarding copyright ownership.
    The ASF licenses this file to You under the Apache License, Version 2.0
    (the "License"); you may not use this file except in compliance with
    the License.  You may obtain a copy of the License at

         http://www.apache.org/licenses/LICENSE-2.0

    Unless required by applicable law or agreed to in writing, software
    distributed under the License is distributed on an "AS IS" BASIS,
    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
    See the License for the specific language governing permissions and
    limitations under the License.

-->
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
    <modelVersion>4.0.0</modelVersion>

    <parent>
        <groupId>org.apache.camel</groupId>
        <artifactId>components</artifactId>
        <version>3.20.0-SNAPSHOT</version>
    </parent>

    <artifactId>camel-iota</artifactId>
    <packaging>jar</packaging>
    <name>Camel :: IOTA (deprecated)</name>
    <description>Camel IOTA component</description>

    <properties>
    </properties>

    <dependencies>

        <dependency>
            <groupId>org.apache.camel</groupId>
            <artifactId>camel-support</artifactId>
        </dependency>

        <!-- IOTA JAVA LIB dependency -->
        <dependency>
            <groupId>org.iota</groupId>
            <artifactId>jota</artifactId>
            <version>${jota-version}</version>
            <exclusions>
                <exclusion>
                    <groupId>ch.qos.logback</groupId>
                    <artifactId>*</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>org.iota</groupId>
                    <artifactId>java-md-doclet</artifactId>
                </exclusion>
            </exclusions>
        </dependency>

        <!-- test dependencies -->
        <dependency>
            <groupId>org.apache.camel</groupId>
            <artifactId>camel-test-spring-junit5</artifactId>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>org.apache.logging.log4j</groupId>
            <artifactId>log4j-slf4j-impl</artifactId>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>org.junit.jupiter</groupId>
            <artifactId>junit-jupiter</artifactId>
            <scope>test</scope>
        </dependency>
    </dependencies>
</project>
	at org.openrewrite.maven.MavenParser.parseInputs(MavenParser.java:132)
	at org.openrewrite.Parser.parse(Parser.java:41)
	at org.openrewrite.maven.MavenMojoProjectParser.parseMaven(MavenMojoProjectParser.java:309)
	at org.openrewrite.maven.MavenMojoProjectParser.listSourceFiles(MavenMojoProjectParser.java:119)
	at org.openrewrite.maven.AbstractRewriteMojo.listResults(AbstractRewriteMojo.java:209)
	at org.openrewrite.maven.AbstractRewriteRunMojo.execute(AbstractRewriteRunMojo.java:48)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:141)
	... 16 common frames omitted
Caused by: org.openrewrite.maven.MavenDownloadingExceptions: null
	at org.openrewrite.maven.MavenDownloadingExceptions.append(MavenDownloadingExceptions.java:44)
	at org.openrewrite.maven.tree.MavenResolutionResult.resolveDependencies(MavenResolutionResult.java:184)
	at org.openrewrite.maven.MavenParser.parseInputs(MavenParser.java:129)
	... 22 common frames omitted
	Suppressed: org.openrewrite.maven.MavenDownloadingException: org.iota:java-md-doclet:2.2 failed. Unable to download POM. Tried repositories:
https://repo1.maven.org/maven2/: HTTP 404
https://packages.atlassian.com/maven-external: HTTP 404
https://repository.jboss.org/nexus/service/local/repositories/thirdparty-releases/content/: HTTP 404
https://repo.maven.apache.org/maven2: HTTP 404
		at org.openrewrite.maven.internal.MavenPomDownloader.download(MavenPomDownloader.java:569)
		at org.openrewrite.maven.tree.ResolvedPom.resolveDependencies(ResolvedPom.java:579)
		at org.openrewrite.maven.tree.ResolvedPom.resolveDependencies(ResolvedPom.java:502)
		at org.openrewrite.maven.tree.MavenResolutionResult.resolveDependencies(MavenResolutionResult.java:179)
		... 23 common frames omitted
```
This looks weird as the `org.iota:java-md-doclet:2.2` has been explicitly excluded from the list of transitive dependencies.  Is that supported by `rewrite-maven` ?

--- a/rewrite-maven/src/main/java/org/openrewrite/maven/MavenDownloadingExceptions.java
+++ b/rewrite-maven/src/main/java/org/openrewrite/maven/MavenDownloadingExceptions.java
@@ -43,6 +43,7 @@ public class MavenDownloadingExceptions extends Exception {
         if (current == null) {
             current = new MavenDownloadingExceptions();
         }
+        current.addSuppressed(exception);
         current.exceptions.add(exception);
         return current;
     }
@@ -52,6 +53,7 @@ public class MavenDownloadingExceptions extends Exception {
         if (current == null) {
             current = new MavenDownloadingExceptions();
         }
+        exceptions.getExceptions().forEach(current::addSuppressed);
         current.exceptions.addAll(exceptions.getExceptions());
         return current;
     }

--- a/rewrite-maven/src/main/java/org/openrewrite/maven/UncheckedMavenDownloadingException.java
+++ b/rewrite-maven/src/main/java/org/openrewrite/maven/UncheckedMavenDownloadingException.java
@@ -24,13 +24,19 @@ public class UncheckedMavenDownloadingException extends RuntimeException {
 
     public UncheckedMavenDownloadingException(Xml.Document pom, MavenDownloadingExceptions e) {
         super("Failed to download dependencies for " + pom.getSourcePath() + ":\n" +
-              e.warn(pom).printAllTrimmed(), e);
-        this.pomWithWarnings = e.warn(pom);
+              warn(pom, e).printAllTrimmed(), e);
+        this.pomWithWarnings = warn(pom, e);
     }
 
     public UncheckedMavenDownloadingException(Xml.Document pom, MavenDownloadingException e) {
-        super("Failed to download dependencies for " + pom.getSourcePath() + ":\n" +
-              MavenDownloadingExceptions.append(null, e).warn(pom).printAllTrimmed(), e);
-        this.pomWithWarnings = MavenDownloadingExceptions.append(null, e).warn(pom);
+        this(pom, MavenDownloadingExceptions.append(null, e));
+    }
+
+    private static Xml.Document warn(Xml.Document pom, MavenDownloadingExceptions mde) {
+        try {
+            return mde.warn(pom);
+        } catch (Exception e) {
+            return pom;
+        }
     }
 }

